src/OSPSuite.Infrastructure.Import/Core/Mappers/DataSetToDataRepositoryMapper.cs

```diff
@@ -97,12 +97,20 @@ private void convertParsedDataColumn(DataRepository dataRepository, KeyValuePair
          //loop over view rows to get the sorted values.
          foreach (var value in column.Value)
          {
-            if (value == null || double.IsNaN(value.Value)) //but we actually should not be allowing this at all right?
+            var adjustedValue =
+               (value == null || (double.IsNaN(value.Value) && double.IsNaN(value.Lloq))) ?
+               double.NaN :
+               (
+                  double.IsNaN(value.Value) || (value.Value < value.Lloq) ?
+                  value.Lloq / 2 :
+                  value.Value
+               );
```

Comment:
It seems it will crash if `value.Value` is **not NaN** but `value.Lloq `**is NaN**?
Also I would move this into a small function; with all those nested ifs is a bit hard to understand what happens.

Reply:
@Yuri05 Should make more sense now